hack/generate.sh

```diff
@@ -36,20 +37,42 @@ virtapi_version=$(getVersion ".VirtApiSha")
 virtcontroller_version=$(getVersion ".VirtControllerSha")
 virthandler_version=$(getVersion ".VirtHandlerSha")
 virtlauncher_version=$(getVersion ".VirtLauncherSha")
+virtoperator_version=$(getVersion ".VirtOperatorSha")
 
 # used as env var for operator
 function getShasum() {
     echo "{{if $1}}@{{$1}}{{end}}"
 }
+
+# without the '@' symbole used in 'getShasum'
+function getRawShasum() {
+    echo "{{if $1}}{{$1}}{{end}}"
+}
 virtapi_sha=$(getShasum ".VirtApiSha")
 virtcontroller_sha=$(getShasum ".VirtControllerSha")
 virthandler_sha=$(getShasum ".VirtHandlerSha")
 virtlauncher_sha=$(getShasum ".VirtLauncherSha")
 
+virtapi_rawsha=$(getRawShasum ".VirtApiSha")
+virtcontroller_rawsha=$(getRawShasum ".VirtControllerSha")
+virthandler_rawsha=$(getRawShasum ".VirtHandlerSha")
+virtlauncher_rawsha=$(getRawShasum ".VirtLauncherSha")
+
 ${KUBEVIRT_DIR}/tools/resource-generator/resource-generator --type=virt-api --namespace={{.Namespace}} --repository={{.DockerPrefix}} --version="$virtapi_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} >${KUBEVIRT_DIR}/manifests/generated/virt-api.yaml.in
 ${KUBEVIRT_DIR}/tools/resource-generator/resource-generator --type=virt-controller --namespace={{.Namespace}} --repository={{.DockerPrefix}} --version="$virtcontroller_version" --launcherVersion="$virtlauncher_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} >${KUBEVIRT_DIR}/manifests/generated/virt-controller.yaml.in
 ${KUBEVIRT_DIR}/tools/resource-generator/resource-generator --type=virt-handler --namespace={{.Namespace}} --repository={{.DockerPrefix}} --version="$virthandler_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} >${KUBEVIRT_DIR}/manifests/generated/virt-handler.yaml.in
 
+# The generation code for CSV requires a valid semver to be used.
+# But we're trying to generate a template for a CSV here from code
+# rather than an actual usable CSV. To work around this, we set the
+# versions to something absurd and do a find/replace with our templated
+# values after the file is generated.
+_fake_replaces_csv_version="1111.1111.1111"
+_fake_csv_version="2222.2222.2222"
+${KUBEVIRT_DIR}/tools/csv-generator/csv-generator --namespace={{.Namespace}} --dockerPrefix={{.DockerPrefix}} --operatorImageVersion="$virtoperator_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} --apiSha="$virtapi_rawsha" --controllerSha="$virtcontroller_rawsha" --handlerSha="$virthandler_rawsha" --launcherSha="$virtlauncher_rawsha" --kubevirtLogo={{.KubeVirtLogo}} --csvVersion="$_fake_csv_version" --replacesCsvVersion="$_fake_replaces_csv_version" --csvCreatedAtTimestamp={{.CreatedAt}} --kubeVirtVersion={{.DockerTag}} >${KUBEVIRT_DIR}/manifests/generated/operator-csv.yaml.in
+sed -i "s/$_fake_csv_version/{{.CsvVersion}}/g" ${KUBEVIRT_DIR}/manifests/generated/operator-csv.yaml.in
```

Comment:
What is going to render the true CSV out of this template, filling the real value for `{{.CsvVersion}}`?
Can whatever that is doing it, simply pass `--csv-version` as an argument to `csv-generator` ?

Reply:
KubeVirt's `hack/build-manifests.sh` fills in the variables.

This isn't a model for how HCO needs to integrate with csv-generator necessarily though. Like my previous comment says, I envision the HCO using csv-generator with real arguments and not messing with all this templating logic like we're doing here. We'll see what makes sense for HCO though.